File: spare_parts_shop_flutter/lib/models/bill_item.dart
import 'product.dart';

class BillItem {
  final int id;
  final Product product;
  final int quantity;
  final double price;
  final double gstPercent;
  final double itemTotal;
  final double discount;

  BillItem({
    required this.id,
    required this.product,
    required this.quantity,
    required this.price,
    required this.gstPercent,
    required this.itemTotal,
    required this.discount,
  });

  factory BillItem.fromJson(Map<String, dynamic> json) {
    return BillItem(
      id: json['id'],
      product: Product.fromJson(json['product']),
      quantity: json['quantity'],
      price: (json['price'] as num).toDouble(),
      gstPercent: (json['gstPercent'] as num).toDouble(),
      itemTotal: (json['itemTotal'] as num).toDouble(),
      discount: (json['discount'] as num).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'product': product.toJson(),
      'quantity': quantity,
      'price': price,
      'gstPercent': gstPercent,
      'itemTotal': itemTotal,
      'discount': discount,
    };
  }
}

File: spare-parts-shop/src/main/java/com/spareparts/model/BillItem.java
package com.spareparts.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "bill_items")
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class BillItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "bill_id", nullable = false)
    @JsonIgnore
    private Bill bill;
    
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;
    
    @Column(nullable = false)
    private Integer quantity;
    
    @Column(nullable = false)
    private Double price;
    
    @Column(name = "gst_percent", nullable = false)
    private Double gstPercent;
    
    @Column(name = "item_total", nullable = false)
    private Double itemTotal;
    
    @Column(nullable = false)
    private Double discount = 0.0;
}